lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -694,41 +723,36 @@ void LCIStrategicPlugin::planWhenAPPROACHING(const cav_srvs::PlanManeuversReques
-        //to keep it consistent and probably too detailed for the user to think about, which is why it is not included in the buffer.
-        //Actually including in the buffer doesn't work because it uses that same buffer to check early and late. If buffer is 2s and 
-        //green starts at 10s, it will check +/-2s from 12s. If the buffer was 2.01s and green starts at 10s again, it checks +/-2.01 
-        //from 12.01, so both checks 10s.
-      }
-      else
-      {
-        nearest_green_entry_time_cached_ = nearest_green_entry_time + ros::Duration(config_.green_light_time_buffer);
-        // NOTE: around ending time of green signal duration, this buffer addition might push ET out of green phase.
-        // However, that is okay because those timestamps that fit this criteria is treated non-green by later late/early arrival checks
+        if (nearest_green_signal_start_time == ros::Time(0)) //in tdb
+        {
+          nearest_green_signal_start_time = ros::Time(lanelet::time::toSec(traffic_light->recorded_time_stamps.back().first)); 
+        }
       }
+
+      // If ET is within green or TDB, it should always aim for at least minimum of "start_time of green or tdb + green_buffer" for safety
+
+      nearest_green_entry_time_cached_ = nearest_green_signal_start_time + ros::Duration(config_.green_light_time_buffer + EPSILON);
+      
+      // EPSILON=0.01 is there because if predictState's input exactly falls on ending_time it picks the previous state.
+      //For example, if 0 - 10s is GREEN, and 10 - 12s is YELLOW, checking exactly 10.0s will return GREEN,
+      //but 10.01s will return YELLOW. This 0.01 convention is used throughout the file, so thought it is better
+      //to keep it consistent and probably too detailed for the user to think about, which is why it is not included in the buffer.
+      //Actually including in the buffer doesn't work because it uses that same buffer to check early and late. If buffer is 2s and 
+      //green starts at 10s, it will check +/-2s from 12s. If the buffer was 2.01s and green starts at 10s again, it checks +/-2.01 
+      //from 12.01, so both checks 10s.
+      
     }
 
     nearest_green_entry_time = nearest_green_entry_time_cached_.get();
   }
-  else if (nearest_green_entry_time_cached_) 
-  { // always pick later of buffered green entry time, or earliest entry time
-    nearest_green_entry_time = ros::Time(std::max(nearest_green_entry_time.toSec(), nearest_green_entry_time_cached_.get().toSec()));
-  }
-  
+
   if (nearest_green_entry_time_cached_ && nearest_green_entry_time > nearest_green_entry_time_cached_.get())
   {
     ROS_DEBUG_STREAM("Earliest entry time has gone past the cached green light. nearest_green_entry_time_cached_:" << std::to_string(nearest_green_entry_time_cached_.get().toSec()) << ", and earliest_entry_time: " << std::to_string(earliest_entry_time.toSec()));
```

Comment:
Another case is that the vehicle ET is pushed back to TBD which is pushed back in time due to a new fixed phase. Not critical, but this is another case where nearest_green_entry_time > nearest_green_entry_time_cached_ in UC3.

Reply:
"... has gone past the cashed entering time" is a more accurate description in my opinion.

Reply:
thanks!